### ISSUE TITLE
Different result cmp to bert/extract_features.py

### ISSUE THREAD

**Issue by Damcy** [zhihu/cuBERT#31]
Thanks for your great jobs.
I'am using the same input seq, but get slightly different results from bert/extract_features.py and cuBERT.

eg.
cuBert id
101 6821 3221 6443 1557 102 872 3221 6443 1557 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
cuBert pooled emb
0.826238 -0.021815 -0.124155 0.455085 1.10134 -0.764902 -0.00428823 0.0152181 ...

bert extract features id
101 6821 3221 6443 1557 102 872 3221 6443 1557 102 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
bert extract features pooled emb
0.827281 -0.021752 -0.123682 0.454883 1.100451 -0.765023 -0.004001 0.015624 ...

Any ideas on this inconsistency?

**Comment by levyfan** [comment]
We do not aim at **exactly** the same. There are a lot of reasons may cause the inconsistency:

* The GPU and CPU output are not **exactly** the same. cuBLAS and MKL have different results. 
* The softmax in our CPU implementation is very very simple and different from our GPU impl (GPU impl is same as tensorflow).
* The gelu is changed and we still use the original impl https://github.com/google-research/bert/commit/bee6030e31e42a9394ac567da170a89a98d2062f

**Comment by Damcy** [comment]
@levyfan Thank you. It seems that there are many factors that could lead to the inconsistency. I'll make more test in my project.
